app/models/decision.rb

```diff
@@ -32,9 +32,9 @@ def weight_split_fully_to_variant
   end
 
   def reassign_existing_assignments
-    transaction do
-      bulk_assignment.save!
-      BulkReassignment.create!(assignments: existing_assignments, bulk_assignment: bulk_assignment)
+    bulk_assignment.save!
+    existing_assignments.find_in_batches do |subset|
```

Comment:
any idea how much faster this will be, since it's still wrapped in an outer transaction?

in mysql-land i wouldn't expect a huge performance win, since the transaction log still has to grow so dang much.